CRM/Contribute/BAO/Contribution.php

```diff
@@ -467,14 +467,29 @@ public static function getPaymentProcessorReadyAddressParams($params, $billingLo
    * @return int
    */
   public function getNumTermsByContributionAndMembershipType($membershipTypeID, $contributionID) {
-    $numTerms = CRM_Core_DAO::singleValueQuery("
-      SELECT membership_num_terms FROM civicrm_line_item li
+    $termsDAO = CRM_Core_DAO::executeQuery("
+      SELECT membership_num_terms, qty FROM civicrm_line_item li
       LEFT JOIN civicrm_price_field_value v ON li.price_field_value_id = v.id
       WHERE contribution_id = %1 AND membership_type_id = %2",
       array(1 => array($contributionID, 'Integer'), 2 => array($membershipTypeID, 'Integer'))
     );
+
+    $membershipTerms = 1;
+
+    while ($termsDAO->fetch()) {
+      $qty = $termsDAO->qty;
+      $numTerms = $termsDAO->membership_num_terms;
+
+      if (!empty($qty)) {
+        $membershipTerms = $qty;
+      }
+      elseif (empty($numTerms)) {
```

Comment:
should this be !empty? @jusfreeman

Reply:
@seamuslee001 Yes it should be **!empty**. Fixed and pushed the changes.